### DISCUSSION TITLE
Time Series Graph: Hover Tooltip Sort Order

### DISCUSSION THREAD

**Discussion by k-bx** [grafana/grafana#38861] [Dashboards, Panels & Visualizations]
There's one very useful feature missing from the new Time Series Graph that is present in the "Graph (old)" which is ability to set decreasing sort order for values under a hover tooltip.

That is a very important and often-used feature for us and it would be nice to see in the new Time Series Graph.

<img width="350" alt="Screenshot 2021-09-05 at 10 45 17" src="https://user-images.githubusercontent.com/107968/132119579-b05e06e8-7448-4546-b40c-385023013872.png">

**Comment by ngocvo3103** [comment]
Got same issue! The only reason I keep coming back to the old graph is this option. Please fix!

**Comment by Mario-Hofstaetter** [comment]
I was in the process of migrating dashboards to `TimeSeries` when I noticed this.
Have to postpone until this is implemented :-/

It looks like the prometheus `sort_desc` function could provide a workaround, but I'm not entirely sure it is useable in all cases, and would mean all dashboard queries that currently have sorted tooltips (from `Graph` panel) would have to be edited.

See also: https://github.com/grafana/grafana/discussions/38305

**Reply by CompPhy** [reply]
In our case we're not using Prometheus, so that's not even an option.  In my mind, this should support sorting, etc. just like the Graph(old) panel.  I get that TimeSeries is trying to do more, which is fine, but it's also advertised as a direct replacement for Graph(old).  If we are going to make such claims, then it should support features that are commonly used.

**Reply by JokerQueue** [reply]
Prometheus sort_desc function doesn't work in my cases in the latest version 8.3.0, the order of the tooltips is fixed. 
query sample:
`sort_desc(sum(rate(elapsed_histogram_count{}[$__rate_interval])) by (app) > 0)`

**Comment by CompPhy** [comment]
Another thought here... We have a couple of graphs where we're overlaying a lot of data points for comparison.  In addition to sorting the tool tip, maybe something like "show the top X items" could be a good option too.  We have one graph where the tool tip can't even fit on the screen and it freaks out trying to display everything.  If we could sort, and choose to show the top X entries, this would help with such issues.

**Reply by Mario-Hofstaetter** [reply]
@CompPhy  Limiting the entries in tooltip is related to issue #18425 (could not find any other / better issue), which has unfortunately been closed without transfer to gh discussion :-/ So maybe this needs a separate gh discussion.
Limiting the amount of series by using the (prometheus) function `topk` is just a half-baked workaround too.

I'm a bit suprised this (and the sorting) is not available in the *next generation graph panel* 😢

**Comment by jdmaloney** [comment]
Getting bit by this issue as well myself and am having to pause roll out of the new panel type.  This seems to be a bit of a trend here with some panel types.  The Pie Chart panel lost features as well.  It is discouraging that new panel types that are *replacing* old panel types don't have feature parity before being released.

**Reply by leeoniya** [reply]
> It is discouraging that new panel types that are replacing old panel types don't have feature parity before being released.

to be fair, the old Graph panel did _way_ too much heavy lifting, to the point that it did a lot of things poorly (performance, bar charts, histograms, aggregation, calcs, series x, etc) and was impossible to continue expanding even further in a maintainable way that did not cause additional bugs and :exploding_head:. there was no feasible way to move forward with a grand split-up of that omni-panel while maintaining every single feature across every decoupled panel type...we would still be working on it all with nothing to show if the only route was non-incremental. it's hard to gauge exactly what features/parity must be prioritized in each iteration; every old Graph feature was used by hundreds to hundreds of thousands of users in the 1M+ Grafana ecosystem.

meanwhile, the new panels fix dozens of extremely old bugs and fulfill dozens more of highly requested features that were absent or impossible to solve in Graph. hopefully this is not easily dismissed as something that should have taken a back seat to 100% feature parity. we are well aware of the remaining high-demand parity features/bugs like this one, and while we work on them you're welcome to continue using the old Graph panel if it meets your needs. what i can tell you is that it _will_ be fixed in the near future.

**Comment by JesseRhoads** [comment]
Upvotes all around; please support popup tooltip sorting in the new time series module.
This is entirely a dealbreaker for us, tooltips not being sorted makes them completely unusable with  bunch of series on one graph, so back to the old graph module we go.

**Comment by Willshield** [comment]
This blocks us also from migrating to the new time-series chart. It would be nice to have it soon.

**Comment by voiprodrigo** [comment]
Need it, badly! :)

**Comment by mikasun7** [comment]
Need it.

**Comment by nktl** [comment]
Very much needed here too!

**Comment by korovkin** [comment]
need it, very badly

**Comment by vladst3f** [comment]
needed !

**Comment by gofly** [comment]
needed!

**Comment by dprokop** [comment]
Support for this missing feature is implemented in this PR: https://github.com/grafana/grafana/pull/43615

**Reply by nikobearrr** [reply]
I was looking at issues related to the tooltip and I found this one. Is it on purpose that the default mode is `Single` compared to the old Graph where it was `All`? I would really like if I can make the `All` default (at least for myself)

**Comment by natellium** [comment]
As mentioned above this feature has been implemented https://github.com/grafana/grafana/discussions/38861#discussioncomment-1899864
